### PR TITLE
Go: Add language-specific baseline configuration

### PR DESCRIPTION
**Pull request by owen-mc**

### CODE REVIEW

**Review by smowton — APPROVED** [review]

**Comment by owen-mc** [comment]
@smowton Note that CI failed because it found Go 1.20.5 in the cache and used that 😞 . This is the first time I've seen 1.20.5 since 1.20.6 started being used.

**Review by henrymercer — COMMENTED** [review]
Could we modify these scripts to return a JSON object in all cases (for instance, just an empty array for `paths-ignore`)?  Right now the CLI will tolerate the output, but will raise a warning that the script didn't return a valid baseline configuration.

**Comment by henrymercer** [comment]
> @smowton Note that CI failed because it found Go 1.20.5 in the cache and used that 😞 . This is the first time I've seen 1.20.5 since 1.20.6 started being used.

The toolcache version used in an Actions run on a repo isn't monotonic, and can wobble during the rollout of a new version of the toolcache.  So it might be an idea to set up a specific patch version of Go, if the tests rely on that.

**Comment by owen-mc** [comment]
@henrymercer Sure. Done. Note that I wasn't sure whether to put this in `tool` or `codeql-tools`. I guess the proof of the pudding is in the eating, but I haven't had a chance to test this yet.

**Comment by henrymercer** [comment]
Looking at the Makefile, `codeql-tools` looks like the right place, though we might need to add the scripts to the `CODEQL_TOOLS` variable in the Makefile too.

**Comment by henrymercer** [comment]
I looked at your branch and it looks good, except I think `go/codeql-tools/baseline-config-vendor.json` and `go/codeql-tools/baseline-config-empty.json` are swapped.  Not sure why we can't see the commit in the PR.

File: go/codeql-tools/configure-baseline.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+if [ -f vendor/modules.txt ]; then
+  cat $CODEQL_EXTRACTOR_GO_ROOT/codeql-tools/baseline-config-vendor.json
+else
+  cat $CODEQL_EXTRACTOR_GO_ROOT/codeql-tools/baseline-config-empty.json
```
Comment (henrymercer) [inline go/codeql-tools/configure-baseline.sh:6]:
```suggestion
  cat $CODEQL_EXTRACTOR_GO_ROOT/tools/baseline-config-vendor.json
else
  cat $CODEQL_EXTRACTOR_GO_ROOT/tools/baseline-config-empty.json
```

**Review by henrymercer — APPROVED** [review]
I built an intree dist for Go and tested this on a repo with vendored modules.  After making the modifications below, it worked 🎉:

```
> codeql database init path/to/db --source-root path/to/source --language go -vvv --calculate-language-specific-baseline
...
Ignored an additional 2516 files when processing baseline information for Go due to paths and paths-ignore configuration.
Found 1254 baseline files for go.
...
```

**Review by henrymercer — APPROVED** [review]

**Comment by owen-mc** [comment]
The test failures were unrelated.

**Pull request merged**